bluesky/callbacks/broker.py

```diff
@@ -27,6 +28,9 @@ def event(self, doc):
             if doc.get('filled', {}).get(field):
                 pass
             else:
+                if self.fs is None:
+                    raise RuntimeError('Either the data must be pre-loaded or'
```

Comment:
Note that this only errors if we need to go looking for a filestore resource (it is not filled) and don't have a filestore instance, which I agree is an undue burden on the users.
I don't see a good way out of this one. Our options (as I see it):
1. Assume that the events are never filled and require a FileStore instance (which breaks if any of the events are filled as it tries to use a numpy array as a key, eg any `hdr.stream(fill=True)`)
1. Assume that the events are always filled and don't require a FileStore instance (which breaks all the existing callbacks at the beamlines as there is no way to fill the events before they reach this callback (at least without some sort of pipeline before the callback))
1. Assume neither and figure it out when we get the events, which may or may not require a FileStore instance. This is as implemented.
1. Split the codebase into two sets of callbacks, one that operates on filled events, the other which does not (which I imagine would become a nightmare to maintain).

Personally I like option 2. as it separates all the database/retrieval part of the code from the data visualization code, but I'm not certain that it is a good idea to break most of the callbacks as they are currently implemented at the beamlines.